sdks/wasm/framework/src/WebAssembly.Bindings/Core/Float32Array.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+namespace WebAssembly.Core {
+	public sealed class Float32Array : TypedArray<Float32Array, float> {
+		public Float32Array () { }
+
+		public Float32Array (int length) : base (length) { }
+
+
+		public Float32Array (ArrayBuffer buffer) : base (buffer) { }
+
+		public Float32Array (ArrayBuffer buffer, int byteOffset) : base (buffer, byteOffset) { }
+
+		public Float32Array (ArrayBuffer buffer, int byteOffset, int length) : base (buffer, byteOffset, length) { }
+
+		internal Float32Array (IntPtr js_handle) : base (js_handle) { }
+
+
+		// Define the indexer to allow client code to use [] notation.
+		public float? this [int i] {
```

Comment:
do we need to duplicate all the array logic like this? i'm guessing there's some reason why we can't use generics

Reply:
The problem was the conversion.  The return values are int or double and were causing a problem with a way to unbox them.

Reply:
I see, makes sense. Can you hoist all the logic except the cast down into a function in the base class? Not a requirement to merge, but would cut out a lot of duplication.

Reply:
Yes, I agree.  I was planning on re-visiting that part as well.  Just wanted to get this out here for review.

Reply:
Sure thing. Not something I consider a merge-blocker, anyway. Big PR so I haven't looked at the whole thing yet but it looks good so far.

Reply:
Array logic moved.